ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -373,10 +376,11 @@ private void updateSealedStateOfReplicas(List<InstanceConfig> instanceConfigs) {
     @Override
     public void onLiveInstanceChange(List<LiveInstance> liveInstances, NotificationContext changeContext) {
       logger.trace("Live instance change triggered from {} with: {}", dcName, liveInstances);
-      if (changeContext.getType() == NotificationContext.Type.INIT) {
+      updateInstanceLiveness(liveInstances);
+      if (!liveStateInitialized) {
```

Comment:
minor: since this one is not under a lock, could you use `AtomicBoolean` and `compareAndSet()`?